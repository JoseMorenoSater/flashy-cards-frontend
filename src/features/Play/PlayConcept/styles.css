.play-lesson-container {
  padding-top: 100px;
  position: absolute;
  height: 80%;
  width: 100%;
}

.height-100-relative {
  height: 100%;
}

@media only screen and (max-height: 400px) {
  .play-lesson-container {
    padding-top: 5%;
    height: 100%;
  }
}

.control-wrapper {
  margin: auto;
}

.control-wrapper > div {
  margin: auto;
}

.control-left {
  width: 0;
  height: 0;
  border-top: 60px solid transparent;
  border-bottom: 60px solid transparent;
  border-right: 60px solid #b6e6bd;
}
.control-left:hover {
  border-right: 60px solid #4baea0;
  cursor: pointer;
}

.control-right {
  width: 0;
  height: 0;
  border-top: 60px solid transparent;
  border-bottom: 60px solid transparent;
  border-left: 60px solid #b6e6bd;
}

.control-right:hover {
  border-left: 60px solid #4baea0;
  cursor: pointer;
}
