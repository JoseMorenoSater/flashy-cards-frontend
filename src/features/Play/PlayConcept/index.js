/**
 *
 * PlayConcept
 *
 */

import React from "react";
import PropTypes from "prop-types";
import Card from "../Card";
import "./styles.css";

class PlayConcept extends React.Component {
  constructor(props) {
    super(props);
    this.state = { cardFlipped: false };
  }

  componentDidUpdate({ cardA }) {
    if (cardA !== this.props.cardA) {
      this.setState({ cardFlipped: false });
    }
  }

  flipCard = () => {
    this.setState({ cardFlipped: !this.state.cardFlipped });
  };

  render() {
    return (
      <div className="play-lesson-container">
        <div className="container height-100-relative">
          <div className="slider row height-100-relative">
            <div className="col-1 col-md-2" onClick={this.props.prevCard}>
              {"<"}
            </div>
            <div className="col-10 col-md-8">
              <Card
                flipCard={this.flipCard}
                cardText={
                  this.state.cardFlipped
                    ? this.props.cardB.text
                    : this.props.cardA.text
                }
              />
            </div>
            <div className="col-1 col-md-2" onClick={this.props.nextCard}>
              {">"}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

PlayConcept.propTypes = {
  cardA: PropTypes.object.isRequired,
  cardB: PropTypes.object.isRequired
};

export default PlayConcept;
