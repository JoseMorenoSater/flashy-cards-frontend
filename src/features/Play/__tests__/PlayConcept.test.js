import React from "react";
import { create } from "react-test-renderer";

import PlayConcept from "../PlayConcept";

describe("PlayConcept Component", () => {
  it("should render correctly", () => {
    const PlayConceptComponent = create(
      <PlayConcept cardA={{}} cardB={{}} />
    ).toJSON();
    expect(PlayConceptComponent).toMatchSnapshot();
  });
});
