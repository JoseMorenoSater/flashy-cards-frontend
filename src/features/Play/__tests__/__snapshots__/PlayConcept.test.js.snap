// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayConcept Component should render correctly 1`] = `
<div>
  This concept is empty
</div>
`;
